### ISSUE TITLE
esp8266 proxy Arduino variant

### ISSUE THREAD

**Issue by danielkucera** [kanflo/opendps#155]
Hi,

what would you say about rewriting esp proxy to Arduino framework in PlatformIO?
I see 3 advantages:
1. easier to build (using PlatformIO)
2. no need to store wifi credentials in binary thanks to: https://github.com/tzapu/WiFiManager
3. easy automated builds: https://docs.platformio.org/en/latest/ci/travis.html

I can try to do this if you help me understand the current code.

**Comment by JackNewman12** [comment]
I agree with all of your points. I enjoy working with PlatformIO and it would make life significantly easier for anyone wanting to work on the esp code.
It also means we could remove the esp-open-rtos submodule since platformIO will handle all the dependencies for us.

**Comment by kanflo** [comment]
I am all for a complete rewrite. It should be rather straight forward. Currently the proxy uses UDP and that needs to be changed for TCP. The proxy does just that, forwards traffic between a socket and a UART.

**Comment by danielkucera** [comment]
Could something like this +- fit? https://github.com/danielkucera/esp8266-arduino-ebus/blob/master/src/main.cpp

**Comment by danielkucera** [comment]
Preliminary version at https://github.com/danielkucera/esp8266-opendps
Also depends on #160

**Comment by danielkucera** [comment]
@kanflo , do you insist on using uHej? I would prefer mDNS, like this:
```
$ ./dpsctl.py -d tcp:esp-opendps.local -o off -q
Func       : cv (off)
  current  : 2000
  voltage  : 5000
V_in       : 32.34 V
V_out      : 0.01 V
I_out      : 0.000 A
```

**Comment by kanflo** [comment]
mDNS is fine, it would be a wasted effort to port uHej to Arduino ;)
